discovery-infra/assisted_service_api.py

```diff
@@ -134,11 +134,11 @@ def generate_and_download_image(self, cluster_id, ssh_key, image_path):
         self.generate_image(cluster_id=cluster_id, ssh_key=ssh_key)
         self.download_image(cluster_id=cluster_id, image_path=image_path)
 
-    def set_hosts_roles(self, cluster_id, hosts_with_roles):
+    def set_hosts_roles(self, cluster_id, hosts_with_roles, hosts_names=None):
```

Comment:
why we have host_names here?

Reply:
The API call in this function can update host names along with roles. For now we need it as a workaround. In all other scenarios it will work as before, updating just the roles.

Reply:
in this case u need to give it another name. Setting role != changing hostname